src/lib/crypto/aes/keys.ts

```diff
@@ -0,0 +1,65 @@
+import { webcrypto } from 'one-webcrypto'
+import utils, { normalizeUnicodeToBuf } from '../utils'
+import { DEFAULT_SYMM_ALG, DEFAULT_SYMM_LEN, DEFAULT_HASH_ALG, DEFAULT_CHAR_SIZE } from '../constants'
+import { SymmKey, SymmKeyOpts, HashAlg, CharSize, ExportKeyFormat } from '../types'
+
+export async function makeKey(opts?: Partial<SymmKeyOpts>): Promise<SymmKey> {
+  return webcrypto.subtle.generateKey(
+    {
+      name: opts?.alg || DEFAULT_SYMM_ALG,
+      length: opts?.length || DEFAULT_SYMM_LEN,
+    },
+    true,
+    ['encrypt', 'decrypt']
+  )
+}
+
+export async function deriveKey(
+  passphrase: string, 
+  salt?: Uint8Array, 
+  opts?: Partial<SymmKeyOpts>,
+  hashAlg: HashAlg = DEFAULT_HASH_ALG,
+  // charSize: CharSize = DEFAULT_CHAR_SIZE
+): Promise<SymmKey> {
+  const enc = new TextEncoder()
+  let baseKey = await webcrypto.subtle.importKey(
+    'raw', // raw format of the key - should be Uint8Array
+    enc.encode(passphrase),
+    'PBKDF2', // Re-use the same algorithm as in encrypt()
+    false, // extractable (we don't need to extract it, just the derived key)
+    ['deriveBits', 'deriveKey'] // what this key can do
+  );
+  // Generate the new key from the password key
+  const alg: Pbkdf2Params = {
+    name: 'PBKDF2',
+    salt: salt ?? new Uint8Array(16),
```

Comment:
Does this have the potential to create a salt that is an all-zero array? We should make sure that is never possible.

Reply:
I'll do this + store the salt in firestore

Reply:
Done. Salt's now randomly generated and stored in Firestore.